reactor-core/src/test/java/reactor/core/publisher/FluxConcatMapTest.java

```diff
@@ -1226,4 +1227,46 @@ public void discardDelayedOnDrainMapperError() {
 		            .verifyThenAssertThat()
 		            .hasDiscardedExactly(1);
 	}
+
+	@Test
+	public void noRequestBeforeOnCompleteWithZeroPrefetch() {
```

Comment:
would it make sense to have additional tests for the concrete class in a separate test case, on top of that one (which tests the API goes down the `FluxConcatMapNoPrefetch` path I guess) and the stress test ? There might not be much left to test other than the `Scannable`, that said, but still... this test can actually be done in the separate test class as well.

Reply:
I actually ran (locally) `FluxConcatMapTest` with `*NoPrefetch` (by adding `prefetch == 0 || true` to `Flux.java`) and it passed all but the prefetch ones.

Glad that you asked, because I wanted to discuss the testing approach here. We could reuse the same set of tests by replacing every `.concatMap` with `.as(this.concatMap(...))`. This will be a massive change in this PR but IMO worth it. WDYT?

Reply:
are you talking about mutualizing the test code, or duplicating the relevant tests? I would definitely agree on duplicating, but if reusing I'm expecting only a subset of the tests make sense for `FluxConcatMapNoPrefetch`?

Reply:
or, actually, an easier way would be to make the prefetch in the test explicit... yes, I think that's the way to go :) Will do.